### ISSUE TITLE
multiple-devices are not cleared properly

### ISSUE THREAD

**Issue by umlaeute** [pure-data/pure-data#2017]
while staring at a fix for #2005, i noticed weird behaviours with *multiple devices*

1. Open MIDI-preferences
2. Enable *Multiple Devices*
3. Select some output-device Nr.1, e.g. `Wine midi mapper`
4. Select some output-device Nr.2, e.g. `Midi Through Port-0`
5. <kbd>Apply</kbd> settings
6. <kbd>Save all settings</kbd>
7. Re-start Pd, and Open MIDI-preferences again
8. Notice that you got 2 output devices, as selected :tada: 
9. Set the 2nd output device to `none`
10. <kbd>Apply</kbd> settings
11. <kbd>Save all settings</kbd>
12. Re-start Pd, and Open MIDI-preferences again
13. Notice that you *still* got 2 output devices, as selected in 3 & 4 :frowning_face: 

I can change the 2nd port to some valid device, and it works as expected.
however, it seems i cannot clear the 2nd device any more.

this seems to be a problem with Pd-core, as the GUI sends a proper midi-settings message
```
pd midi-dialog  3 0 0 0 0 0 0 0 0  1 0 0 0 0 0 0 0 0  4 5;
pd save-preferences;
```

However, the `.pdsettings` (registry in Windows) still shows both devices:

```
midioutdev1        0
midioutdev2        3
midioutdevname1    Wine midi mapper
midioutdevname2    Midi Through Port-0
```

I'm testing with Pd-0.53-0 on Wine64.
So this is *not a regression*, but existing (bad) behavior (and I don't think it is Wine specific; ~~or Windows specific - but that is currently the simplest platform for me to test~~)

**Comment by umlaeute** [comment]
also note that I checked whether things work better with the Audio preferences, and it seems to has the same problem in the `.pdsettings` (registry).
however since the number of channels is stored along the device index, Pd has some additional notion that the 2nd audio device is "disabled" and doesn't show up in the preference window...

**Comment by umlaeute** [comment]
ah, the problem comes most likely from the fact, that on Linux we just create a new `~/.pdsettings` file ditching the old one (and I think on macOS we are basically doing the same), whereas on Windows we write to the registry without clearing old values.

we cannot simply delete the `HKCU\Software\Pure-Data` entry, because this is the root-node for other prefs (most notably, the GUI-preferences that live in  `HKCU\Software\Pure-Data\org.puredata.pd.pd-gui` ).

**Comment by Spacechild1** [comment]
I guess we just have to wipe the `{audio/midi}{in/out}dev[name]N` entries. I will try a fix and push to develop.

**Comment by Spacechild1** [comment]
Should be fixed with 6c7cc469b2af4a7bf03db35945f3a49dbc40c13a
